halo2_proofs/src/dev.rs

```diff
@@ -746,8 +746,26 @@ impl<F: FieldExt> MockProver<F> {
     /// Returns `Ok(())` if this `MockProver` is satisfied, or a list of errors indicating
     /// the reasons that the circuit is not satisfied.
     pub fn verify(&self) -> Result<(), Vec<VerifyFailure>> {
+        self.verify_at_rows(self.usable_rows.clone())
+    }
+
+    /// Returns `Ok(())` if this `MockProver` is satisfied, or a list of errors indicating
+    /// the reasons that the circuit is not satisfied.
+    /// Constraints and lookups are only checked at `row_ids`.
+    pub fn verify_at_rows<I: Iterator<Item = usize>>(
+        &self,
+        row_ids: I,
+    ) -> Result<(), Vec<VerifyFailure>> {
         let n = self.n as i32;
 
+        let row_ids: Vec<_> = row_ids.collect();
```

Comment:
Why collecting directly here? 
You can use the row_ids iter and zip or chain it when needed and use `copied` or `cloned` depending on your needs.

Reply:
make sense. i will modify this